src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1472,7 +1472,6 @@ private void ReportNullabilityMismatchWithTargetDelegate(SyntaxNode syntax, Name
                         }
                         else
                         {
-                            Debug.Assert(sourceTypeOpt?.IsReferenceType != true);
                             return null;
```

Comment:
>return null; [](start = 28, length = 12)

Why is null the right return value in case when the assert fails?

Reply:
Updated.